internal/pkg/service/buffer/store/watcher.go

```diff
@@ -54,51 +50,48 @@ func NewWatcher(ctx context.Context, store *Store) (*Watcher, error) {
 			if err != nil {
 				return nil, err
 			}
-			w.slicesForExports[export.ExportKey] = slice.SliceID
+			w.slicesForExports.Store(export.ExportKey, &slice.SliceID)
 		}
 	}
 
 	return w, nil
 }
 
-func (w *Watcher) GetMappings(key key.ReceiverKeyWithSecret) (map[key.ExportKey]*model.Mapping, bool) {
-	mappings, found := w.mappings[key]
+func (w *Watcher) GetMappings(k key.ReceiverKey) (map[key.ExportKey]*model.Mapping, bool) {
+	mappings, found := w.mappings.Load(k)
 	if !found {
 		return nil, false
 	}
-	return mappings, true
+	return mappings.(map[key.ExportKey]*model.Mapping), true
 }
 
-func (w *Watcher) GetSliceID(key key.ExportKey) time.Time {
-	return w.slicesForExports[key]
+func (w *Watcher) GetSecret(k key.ReceiverKey) (string, bool) {
+	secret, found := w.secrets.Load(k)
+	if !found {
+		return "", false
+	}
+	return secret.(string), true
 }
 
-func (w *Watcher) Watch() {
-	/* output := make(chan struct{}, 1)
+func (w *Watcher) GetSliceID(key key.ExportKey) (*time.Time, bool) {
+	sliceID, found := w.slicesForExports.Load(key)
+	if !found {
+		return nil, false
+	}
+	return sliceID.(*time.Time), true
+}
 
-	go func() {
-		chCtx, cancel := context.WithCancel(ctx)
-		ch := client.Watch(ctx, "", etcd.WithFragment(), etcd.WithProgressNotify(), etcd.WithCreatedNotify())
-		for resp := range ch {
-			if resp.Err() != nil {
-				// error processing
-				break
-			}
-			select {
-			case <-chCtx.Done():
-			case output <- struct{}{}:
-			}
-		}
-		cancel()
-	}()
+func (w *Watcher) Watch(_ context.Context, _ etcd.Client) {
+	// TODO
+}
 
-	for {
-		select {
-		case <-ctx.Done():
-			return
-		case <-output:
-			// some logic code here
-			fmt.Printf(">>> %#v\n", output)
-		}
-	} */
+func (w *Watcher) addMappingToExport(recKey key.ReceiverKey, expKey key.ExportKey, mapping *model.Mapping) {
+	w.lock.Lock()
+	exportMappings := make(map[key.ExportKey]*model.Mapping)
+	if em, found := w.mappings.Load(recKey); found {
+		exportMappings = em.(map[key.ExportKey]*model.Mapping)
+	}
+	exportMappings[expKey] = mapping
+	w.mappings.Store(recKey, exportMappings)
+	w.lock.Unlock()
```

Comment:
@michaljurecko Tady si nejsem moc jistej. Použil jsem `sync.Map` pro data co tu potřebuju ukládat, t.j. i pro `w.mappings`, jenže to je nested map, kterou tady potřebuju updatovat. Když se na to teď dívám s čerstvýma očima, tak si říkám že tady ten zámek je asi blbost. 🤔 Asi bych to teda udělal jako nested sync.Mapy, nebo to taky nepomůže? Když tady načtu mapu, přidám do ní value a mezi tím než to zase uloží z ní někdo něco přečte, tak to asi neva, update prostě ještě nedoběhl.(?)

Reply:
K `map` sa vzdy pristupuje cez referenciu, teda si mozes predstavit, ze je to vzdy pointer.
Map sa da komplet prepisat/inicializovat iba pomocou `make` (alebo inline `m := map[foo]bar{....}` (co ty nemas).

To ci je nested alebo nie nehra ulohu, proste si si tu referenciu na map ulozil do inej map, takze to mozes z uvazovania vyhodit.

Teda ked vsade pouzijes `sync.Map`, tak by si nemal mat problem.

Jedine na co si musis dat pozor je toto:
- ak zistis, ze dany kluc v `map` nie je, a musis inicializovat tu nested map
- tak to musis urobit cez https://pkg.go.dev/sync#Map.LoadOrStore 
- kedze za ten cas co si zistil, ze tam nie je kluc a jeho ulozenim, tam moze ulozit nieco niekto iny
- a ty by si to prepisal s prazdnou map
- .... jedine, ze by si mal nejak inak garantovane, ze tam nikto iny nezapise
- napr. ze sa do danej map zapisuje iba na jednom mieste v kode a nie paralelne

-------------

Neviem ci to takto staci, nepozrel som PR, takze iba takto vseobecne (zatial).

Reply:
💡Ahá no jasně. Díky